include/swift/Driver/Job.h

```diff
@@ -230,9 +230,17 @@ class CommandOutput {
 class Job {
 public:
   enum class Condition {
+    // There was no input map, or the map makred this Job as dirty
+    // but the input didn't change.
+    // Be maximally conservative with dependencies.
     Always,
+    // The input changed, or this job was private or [it was not dirty but
+    // primary output was missing].
```

Comment:
Let's stick to "non-cascading" as the terminology, but even then it's something more like "this job was scheduled as non-cascading in the last build but didn't get to run".

Reply:
Changed the term "private". I didn't want to remove the lower-level predicate, so I added your sentence to the comment.